pkg/scheduler/ugm/queue_tracker.go

```diff
@@ -31,102 +29,197 @@ import (
-		zap.String("application", applicationID),
-		zap.Stringer("resource", usage),
-		zap.Stringer("total resource after decreasing", qt.resourceUsage),
-		zap.Int("total applications after decreasing", len(qt.runningApplications)))
-
 	childQueuePath, immediateChildQueueName := getChildQueuePath(queuePath)
 	if childQueuePath != "" {
 		if qt.childQueueTrackers[immediateChildQueueName] != nil {
-			removeQT, err := qt.childQueueTrackers[immediateChildQueueName].decreaseTrackedResource(childQueuePath, applicationID, usage, removeApp)
-			if err != nil {
-				return false, err
+			removeQT, decreased := qt.childQueueTrackers[immediateChildQueueName].decreaseTrackedResource(childQueuePath, applicationID, usage, removeApp)
+			if !decreased {
+				return false, decreased
 			}
 			if removeQT {
+				log.Log(log.SchedUGM).Debug("Removed queue tracker linkage from its parent",
+					zap.String("queue path ", queuePath),
+					zap.String("removed queue name", immediateChildQueueName),
+					zap.String("parent queue", qt.queueName))
 				delete(qt.childQueueTrackers, immediateChildQueueName)
 			}
 		} else {
 			log.Log(log.SchedUGM).Error("Child queueTracker tracker must be available in child queues map",
 				zap.String("child queueTracker name", immediateChildQueueName))
-			return false, fmt.Errorf("child queueTracker tracker for %s is missing in child queues map", immediateChildQueueName)
+			return false, false
 		}
 	}
 
+	qt.resourceUsage.SubFrom(usage)
+	if removeApp {
+		log.Log(log.SchedUGM).Debug("Removed application from running applications",
+			zap.String("application", applicationID),
+			zap.String("queue path", queuePath),
+			zap.String("queue name", qt.queueName))
+		delete(qt.runningApplications, applicationID)
+	}
+	log.Log(log.SchedUGM).Debug("Successfully decreased resource usage",
+		zap.String("queue path", queuePath),
+		zap.String("application", applicationID),
+		zap.Stringer("resource", usage),
+		zap.Stringer("total resource after decreasing", qt.resourceUsage),
+		zap.Int("total applications after decreasing", len(qt.runningApplications)))
+
 	// Determine if the queue tracker should be removed
-	removeQT := len(qt.childQueueTrackers) == 0 && len(qt.runningApplications) == 0 && resources.IsZero(qt.resourceUsage)
-	return removeQT, nil
+	removeQT := len(qt.childQueueTrackers) == 0 && len(qt.runningApplications) == 0 && resources.IsZero(qt.resourceUsage) &&
+		qt.maxRunningApps == 0 && resources.Equals(resources.NewResource(), qt.maxResources)
+	log.Log(log.SchedUGM).Debug("Can queue tracker be removed?",
+		zap.String("queue path ", queuePath),
+		zap.Int("no. of child queue trackers", len(qt.childQueueTrackers)),
+		zap.Int("no. of running applications", len(qt.runningApplications)),
+		zap.Bool("current resource usage", resources.IsZero(qt.resourceUsage)),
+		zap.Uint64("max running applications", qt.maxRunningApps),
+		zap.String("max resources", qt.maxResources.String()),
+		zap.Bool("remove QT", removeQT))
```

Comment:
Lot of overhead: we know exactly when we need to remove,
Create a single DEBUG log line:
```
log.Log(log.SchedUGM).Debug("Remove queue tracker",
		zap.String("queue path ", queuePath),
		zap.Bool("remove ", removeQT))
```